src/daemon/daemon.cpp

```diff
@@ -624,6 +626,16 @@ mp::MemorySize compute_final_image_size(const mp::MemorySize image_size,
         disk_space = *command_line_value;
     }
 
+    auto available_bytes = QStorageInfo::root().bytesAvailable();
```

Comment:
I see two issues with this:
1. we may be storing images elsewhere, either because multipass is installed in a different filesystem or because [`MULTIPASS_STORAGE`](https://github.com/canonical/multipass/pull/1789) was used
1. this does not currently work under snap confinement; you can try it yourself by using the PR channel which is built in CI: `multipass install/refresh channel=edge/pr1854 multipass`; but we're looking for a way to overcome this and will write back when we find it

Reply:
You also need to cater for [`QStorageInfo::bytesAvailable`](https://doc.qt.io/qt-5/qstorageinfo.html#bytesAvailable) returning `-1` in case of error.

Reply:
@ricab @Saviq Thank you for reviewing this! I changed from checking the root's available space to using `DaemonConfig::data_directory` as @Saviq suggested. I also added a check for if `QStorageInfo::bytesAvailable` returns `-1`. The "BuildAndTest" checks don't seem to be running (nothing has updated after over an hour), so I thought I should just go ahead and let you know about these updates to see what you think.